pandas/core/generic.py

```diff
@@ -4028,6 +4064,10 @@ def rank(self, axis=0, method='average', numeric_only=None,
         Compute numerical data ranks (1 through n) along axis. Equal values are
         assigned a rank that is the average of the ranks of those values
 
+        Do not call this function via the numpy library, as numpy's version
+        has a completely different purpose (returns number of dimensions in
+        an array)!
+
```

Comment:
we have a really different signature as pandas rank is much more friendly to different dtypes, you can prob just raise on this one if someone does ``np.rank(...)`` but if you can't it doesn't really matter. again if someone is doing it, then they must know what they are doing, no amount of documentation will help.